### PR TITLE
Making this package actually an executable

### PR DESCRIPTION
**Pull request by blakeduncan14**

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -78,14 +82,17 @@ const argv = yargs
           return yargs
               .option('token', {
                 alias: 't',
+                demandOption: true,
```
Comment (jakemercurio) [inline src/index.ts]: So this means `token` is required?

File: src/index.ts [hunk 2]
```diff
@@ -63,8 +65,10 @@ async function publishMostRecentBuild(accessToken: string, user: string, siteId:
     const deployId = await getNetlifyDeployId(axiosInstance, siteId, accessToken);
     await publishDeploy(axiosInstance, siteId, deployId, accessToken);
     await lockAutoPublishes(axiosInstance, siteId, deployId, accessToken);
+    process.exit(0);
```
Comment (jakemercurio) [inline src/index.ts]: Does this happen by default?

**Comment by blakeduncan14** [comment]
> Looks good. So in theory I should be able to run `npx netlify-locked-publish` from my terminal once you publish this? Where is the build step, btw? How does the typescript in `src/` get to `lib/` on publish?

@chasdevs the *prepare* script in the package.json gets run before the package is packed and published.  Thats where I added ```npm run build``` which transpiles the code into the lib directory.  Another good script I saw was *prepublishOnly* that runs before the *prepare* script.  I added a lint check there but if I had tests I could have also added a test run.

The "bin" option in the package.json is what creates an executable.  I just set that to the main index.js file because that is the only thing in this app.  But with that added yeah you can run ```npx netlify-locked-publish```.

File: src/index.ts (hunk 1 above)
Comment (blakeduncan14) [reply src/index.ts]: Yeah the access token, user and site are all required to be able to hit netlify's api.

File: src/index.ts (hunk 2 above)
Comment (blakeduncan14) [reply src/index.ts]: I honestly didn't know so I just put it in there.  I can look into the yargs package a little bit more though.

**Pull request merged**